### PR TITLE
Add `unsafe` option

### PR DESCRIPTION
**Pull request by Disorrder**
Add `unsafe` option to `ui-options`.
When `unsafe` set to true, apply value to `$modelValue` even if it isn't valid. By default, if you write invalid value, model stay `undefined`.
Also add `ui-unsafe` attribute, which set bool value to `uiOptions`.

I'm using this option for saving form to draft. When I save form to draft, I don't need validation. But when I save form clearly, I'm checking validation and show message.

Tested on `+7 9999999999` mask. This is phone number. If I missed some digits in number and submit form for draft (with unsafe option), I need save it as-is.

Example (Jade syntax): 
````Jade
form(name="survey")
  input(ng-model="survey.mobile", ui-mask="+7 9999999999", allow-invalid-value)
  input(ng-model="survey.homePhone", ui-mask="+7 9999999999", ui-options="{allowInvalidValue: true}")
````

### CODE REVIEW

File: src/mask.js [hunk 1]
```diff
@@ -144,11 +146,10 @@ angular.module('ui.mask', [])
...
                                 controller.$setValidity('mask', isValid);
-                                if (isValid) {
-                                    return modelViewValue ? controller.$viewValue : value;
-                                } else {
-                                    return undefined;
-                                }
+
+                                if (modelViewValue) value = controller.$viewValue;
+                                if (!value.length) return undefined;
+                                if (isValid || linkOptions.unsafe) return value;
```
Comment (lukepfeiffer10) [inline src/mask.js]: What happens when it's not valid and the `unsafe` attribute is false? I think there is an else block needed to return `undefined` like it does currently.

Comment (Disorrder) [inline src/mask.js]: undefined returns in function by default on function end

File: src/mask.js [hunk 2]
```diff
@@ -188,6 +189,9 @@ angular.module('ui.mask', [])
                             if (angular.isDefined(iAttrs.uiMaskPlaceholderChar)) {
                                 iAttrs.$observe('uiMaskPlaceholderChar', initPlaceholderChar);
                             }
+                            iAttrs.$observe('uiUnsafe', function(unsafe){
```
Comment (lukepfeiffer10) [inline src/mask.js]: I feel like you should check if the attribute is defined before `$observe`ing it similar to the other attribute options.

File: src/mask.js (hunk 1 above)
Comment (lukepfeiffer10) [inline src/mask.js]: Sure but it doesn't hurt to be explicit so it doesn't get lost somewhere in the future if other functionality is added.

**Comment by lukepfeiffer10** [comment]
I would like some unit tests around this feature as well before merging.

Also, not a huge deal but I think it would make more sense if the name was something like `allow-invalid-value`. The term `unsafe` is a bit ambiguous.

**Comment by lukepfeiffer10** [comment]
One more thing. If you are adding something to the `linkOptions` it should also be added to the `uiMaskConfig` so that it can be configured globally as well.

**Comment by Disorrder** [comment]
Okay, then I'll rename. `ui-unsafe` is short and looks like good for attribute, so I choosed it. But `allow-invalid-value` not looks simple, so I better remove this option from attrs

File: src/mask.js (hunk 1 above)
Comment (Disorrder) [inline src/mask.js]: Not sure that duplicating `return undefined` is good. If you want to add any feature here, you'll have to duplicate your feature code.

**Comment by Disorrder** [comment]
Fixed

**Comment by Disorrder** [comment]
Rename in README and to `uiMaskConfig` some later :)

**Comment by lukepfeiffer10** [comment]
@Disorrder Have you worked on implementing some unit tests for this feature and making sure it's added to the `uiMaskConfig` object?

**Comment by Disorrder** [comment]
@lukepfeiffer10 Today I'll do. There was 4 days holidays in Russia :D 
By the way, I have a problem with running gulpfile. Sauce is running with error
`
$ gulp build
module.js:338
    throw err;
    ^
Error: Cannot find module 'sauce-connect-launcher'
` 
No idea what to do

**Comment by lukepfeiffer10** [comment]
Have you made sure to do an `npm install`?

**Comment by Disorrder** [comment]
ofc I have. I've started gulp build successfully when commented 20th string of gulpfile like this
```
var sauceConnectLauncher = null; //require('sauce-connect-launcher');
```
And it builds. But without tests. So, without them, I can't write new test :)

**Comment by lukepfeiffer10** [comment]
I'm not sure how to actually fix your issue but you shouldn't need `sauce-connect-launcher` to run the tests locally so I would just comment it out locally for now to get your tests running and then maybe figure out the module issue later.

**Comment by lukepfeiffer10** [comment]
@Disorrder any updates here?

**Comment by Disorrder** [comment]
@lukepfeiffer10 I didn't forget, just too busy on my job. I think, I'll update today.

**Comment by Disorrder** [comment]
This is blind tests. Please, run them. Hope it works

**Comment by lukepfeiffer10** [comment]
There are some tests failing but they aren't causing the build to fail anymore which is why everything looked good. I'm adjusting some things to hopefully get this working again soon.

**Comment by lukepfeiffer10** [comment]
Released this as 1.8.4

**Comment by ChinthapalliChaitanyaTeja** [comment]
Hi I have used UI-Mask directive to validate US phone format number, it was working fine till version 1.3.0 and from above versions its change the alert...

Can any one help me on this
phone number

**Pull request merged**